### DISCUSSION TITLE
The level of flux model is not match with the data

### DISCUSSION THREAD

**Discussion by kittipongwangnok** [phoebe-project/phoebe2#928] [General]
The shape of light curve look likes the data but the level is not the same, How can we solve this problem?
Thanks
<img width="754" alt="Screenshot 2567-08-24 at 10 19 19 AM" src="https://github.com/user-attachments/assets/1272be76-c94c-4820-9a54-6b493b26696f">

**Comment by kecnry** [comment]
You need to account for the passband luminosity and/or the distance.  Please see the following tutorials and let us know if anything isn't clear:
* [passband luminosity](https://phoebe-project.org/docs/latest/tutorials/pblum)
* [distance](https://phoebe-project.org/docs/latest/tutorials/distance)
* [third light](https://phoebe-project.org/docs/latest/tutorials/l3)
* [Inverse problem example](https://phoebe-project.org/docs/latest/examples/inverse_paper_examples)

**Comment by kittipongwangnok** [comment]
Thank you for the clarification.
However, I got the another problem when I have tried to change the radius
of primary and secondary stars with the error likes this:

ValueError: Could not lookup ld_coeffs for ***@***.***
@***@***.***@dataset.  Try changing ld_coeffs_source to a table that covers a
sufficient range of values or set ld_mode to 'manual' and manually provide
coefficients via ld_coeffs. Enable 'warning' logger to see out-of-bound
arrays.

How can we solve this problem?

Kittipong

On Mon, Aug 26, 2024 at 7:09 PM Kyle Conroy ***@***.***>
wrote:

> You need to account for the passband luminosity and/or the distance.
> Please see the following tutorials and let us know if anything isn't clear:
>
>    - passband luminosity
>    <https://phoebe-project.org/docs/latest/tutorials/pblum>
>    - distance <https://phoebe-project.org/docs/latest/tutorials/distance>
>    - third light <https://phoebe-project.org/docs/latest/tutorials/l3>
>    - Inverse problem example
>    <https://phoebe-project.org/docs/latest/examples/inverse_paper_examples>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/phoebe-project/phoebe2/discussions/928#discussioncomment-10451090>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALO4C3CBQ3CQHCG3S3N2RDTZTMLG5AVCNFSM6AAAAABNBGILP2VHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTANBVGEYDSMA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Kittipong Wangnok
+6684 828 6505
School of Physics
Institute of Science
Suranaree University of Technology

**Reply by kecnry** [reply]
The error message is telling you what to do (effectively the same as your [previous question](https://github.com/phoebe-project/phoebe2/issues/924#issuecomment-2290833199)), and there are also tutorials on these topics on the website that I linked to in the response there.  If you try that and are still running into issues, please let us know!

**Comment by kittipongwangnok** [comment]
I already input this parameter but still get the error with message:
ValueError: Could not lookup ld_coeffs for ***@***.***
@***@***.***  Try changing ld_coeffs_source_bol to a table that covers
a sufficient range of values or set ld_mode_bol to 'manual' and manually
provide coefficients via ld_coeffs_bol. Enable 'warning' logger to see
out-of-bound arrays.

How can I solve this problem

Kittipong

On Mon, Aug 26, 2024 at 7:49 PM Kyle Conroy ***@***.***>
wrote:

> The error message is telling you what to do (effectively the same as your previous
> question
> <https://github.com/phoebe-project/phoebe2/issues/924#issuecomment-2290833199>),
> and there are also tutorials on these topics on the website that I linked
> to in the response there. If you try that and are still running into
> issues, please let us know!
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/phoebe-project/phoebe2/discussions/928#discussioncomment-10451537>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALO4C3G2VLL3PIP5MNL5EETZTMP5PAVCNFSM6AAAAABNBGILP2VHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTANBVGE2TGNY>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Kittipong Wangnok
+6684 828 6505
School of Physics
Institute of Science
Suranaree University of Technology

**Reply by kecnry** [reply]
What have you tried that is still giving this message?

**Comment by kittipongwangnok** [comment]
Changing the radius of r1 and r2?

On Tue, Aug 27, 2024 at 8:00 PM Kyle Conroy ***@***.***>
wrote:

> What have you tried that is still giving this message?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/phoebe-project/phoebe2/discussions/928#discussioncomment-10463165>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALO4C3EJM3JWPKTOMMQL7KDZTRZ7BAVCNFSM6AAAAABNBGILP2VHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTANBWGMYTMNI>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Kittipong Wangnok
+6684 828 6505
School of Physics
Institute of Science
Suranaree University of Technology

**Reply by kecnry** [reply]
What are the masses/radii/temperatures?  Can you enable the warning logger and paste the output of the message that states the out-of-bounds values?

**Comment by kittipongwangnok** [comment]
Since there is no reference of parameters for this system, I have tried to
change the parameters following the example that I have.
Moreover, How can I  enable the warning logger?

Thanks
Kittipong

On Tue, Aug 27, 2024 at 8:07 PM Kyle Conroy ***@***.***>
wrote:

> What are the masses/radii/temperatures? Can you enable the warning logger
> and paste the output of the message that states the out-of-bounds values?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/phoebe-project/phoebe2/discussions/928#discussioncomment-10463239>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALO4C3CKXENHIRZ4ZNSNWPDZTR2YBAVCNFSM6AAAAABNBGILP2VHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTANBWGMZDGOI>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Kittipong Wangnok
+6684 828 6505
School of Physics
Institute of Science
Suranaree University of Technology

**Reply by kecnry** [reply]
All of your messages sent through email are being censored by GitHub (probably thinking that `@` indicates email addresses), can you please go to the page and manually edit the messages so that they are readable?

> How can I  enable the warning logger? 

https://github.com/phoebe-project/phoebe2/issues/924#issuecomment-2293424953

**Comment by kittipongwangnok** [comment]
This is the set of parameters that I have:
b.set_value('incl@binary@component', 85)
b.set_value('q@binary@component', 0.5)
b.set_value('sma@binary@component', 7)
b.set_value('teff@primary@component', 4500)
b.set_value('teff@secondary@component', 6000)
b.set_value('requiv@primary@component', 35)
b.set_value('requiv@secondary@component', 10)
b.set_value('distance@system', 3.03)
b['ld_coeffs_source_bol@primary'] = 'phoenix'  # Or 'ck2004' or 'auto'
b['ld_coeffs_source_bol@secondary'] = 'phoenix'  # Or 'ck2004' or 'auto'
b.set_value('ld_mode', component='primary', dataset='lc01', value='manual')
b.set_value('ld_mode', component='secondary', dataset='lc01', value='manual')
b.set_value('ld_func', component='primary', dataset='lc01', value='logarithmic')
b.set_value('ld_func', component='secondary', dataset='lc01', value='logarithmic')
b.set_value_all('ld_mode_bol', value='manual')
b.set_value('atm@primary@phoebe01@phoebe@compute', value='blackbody')
b.set_value('atm@secondary@phoebe01@phoebe@compute', value='blackbody')
b.set_value('mass@primary@star@constraint', 3.00)
b.set_value('mass@secondary@star@constraint', 1.5)

When I tried to input:
b.set_value('period@binary@component', 310)
b.set_value('t0_supconj@binary@component', 58667.9)

I also get the error.
However, PHOEBE has the limitation of the binary period or not because my system is ~300 days.

**Reply by kecnry** [reply]
Are these the stellar parameters you expect?

```
>>> print(b.filter(qualifier=['mass', 'requiv', 'teff'], context='component'))
ParameterSet: 6 parameters
         requiv@primary@component: 35.0 solRad
           teff@primary@component: 4500.0 K
           mass@primary@component: 3.0 solMass
       requiv@secondary@component: 10.0 solRad
         teff@secondary@component: 6000.0 K
         mass@secondary@component: 1.5 solMass
```

This is showing an error that the secondary is overflowing before even getting to the error message about the atmosphere tables.  Can you please attach a full minimal script to reproduce your case?  If you have real data, you can exclude that by just creating a synthetic light curve instead.

**Reply by kittipongwangnok** [reply]
Thank you for your explanation. 
The set of parameters is just the guess based on the information that I known.
However, I would like to find the best parameters that match to our real observed data.

**Reply by kittipongwangnok** [reply]
This is the example of script and data
[r_times.txt](https://github.com/user-attachments/files/16763867/r_times.txt)

```
import phoebe
from phoebe import u # units
import numpy as np

logger = phoebe.logger('error')

#phoebe.download_passband('Bolometric:900-40000')
#phoebe.download_passband('Johnson:B')

# Initialize the binary system
b = phoebe.default_binary()
b.set_value('incl@binary@component', 78)
b.set_value('q@binary@component', 0.69)
b.set_value('sma@binary@component', 7.5)
b.set_value('teff@primary@component', 10528)
b.set_value('teff@secondary@component', 7970)
b.set_value('requiv@primary@component', 2.61)
b.set_value('requiv@secondary@component', 1.76)
b.set_value('distance@system', 2.45)
b.set_value('gravb_bol@primary@component', 0.93)
b.set_value('gravb_bol@secondary@component', 0.88)
b.set_value('irrad_frac_refl_bol@primary@component', 0.92)
b.set_value('irrad_frac_refl_bol@secondary@component', 0.84)

# Load light curve data
#InputFile = "binned_data_8314879.dat"
InputFile = "r_times.txt"
Data = np.genfromtxt(InputFile)
time  = np.array(Data[:, 0])
flux = np.array(Data[:, 1])
flux_err = np.array(Data[:, 2])

b = phoebe.default_binary()

b.add_dataset('lc', 
              compute_phases=phoebe.linspace(0,1,201),
              times=time, 
              fluxes=flux, 
              sigmas=flux_err, 
              dataset='lc01')

b.add_compute('ellc', compute='fastcompute')
b.set_value_all('ld_mode', 'lookup')
#b.run_compute(compute='fastcompute')

b.set_value('period@binary@component', 310)
b.set_value('t0_supconj@binary@component', 2458667.9)

b.run_compute(compute='fastcompute', model='orig_model')
```

**Reply by kecnry** [reply]
Ok, thanks.  You have a second `b = phoebe.default_binary()` in there that is going to re-set all the changes you made to the system prior to that line.  Removing that and adding a debugging logger exposes the elements that are out-of-bounds:

```
import phoebe

b = phoebe.default_binary()
b.set_value('incl@binary@component', 78)
b.set_value('q@binary@component', 0.69)
b.set_value('sma@binary@component', 7.5)
b.set_value('teff@primary@component', 10528)
b.set_value('teff@secondary@component', 7970)
b.set_value('requiv@primary@component', 2.61)
b.set_value('requiv@secondary@component', 1.76)
b.set_value('distance@system', 2.45)
b.set_value('gravb_bol@primary@component', 0.93)
b.set_value('gravb_bol@secondary@component', 0.88)
b.set_value('irrad_frac_refl_bol@primary@component', 0.92)
b.set_value('irrad_frac_refl_bol@secondary@component', 0.84)


b.add_dataset('lc', 
              compute_phases=phoebe.linspace(0,1,6),
              dataset='lc01')

b.add_compute('ellc', compute='fastcompute')
b.set_value_all('ld_mode', 'lookup')

b.set_value('period@binary@component', 310)
b.set_value('t0_supconj@binary@component', 2458667.9)

phoebe.logger('debug')
b.run_compute(compute='fastcompute', model='orig_model')
```

```
ValueError: Atmosphere parameters out of bounds: ldatm=ck2004, teff=[10528.], logg=[-0.85295305], abun=[0.]
```

and the following print statement shows that the stellar parameters are entirely unphysical (near zero-mass which is resulting in this logg value):

```
>>> print(b.filter(qualifier=['teff', 'mass', 'requiv', 'requiv_max'], context='component'))
ParameterSet: 8 parameters
         requiv@primary@component: 2.61 solRad
C    requiv_max@primary@component: 3.0970514659323665 solRad
           teff@primary@component: 10528.0 K
C          mass@primary@component: 3.485461822028102e-05 solMass
       requiv@secondary@component: 1.76 solRad
C  requiv_max@secondary@component: 2.607248025377602 solRad
         teff@secondary@component: 7970.0 K
C        mass@secondary@component: 2.4049686571993898e-05 solMass
```

My suggestion would be to first work towards understanding the constraints in phoebe (go through the tutorials or examples from past workshops) and ensure that your input parameters are realistic and representative of the type of system you are trying to model.  If your system is physical but still falls outside the existing atmosphere tables (which can happen), then follow the advice in the error messages to use a different atmosphere table (or blackbody, if absolutely necessary).

Alternatively, you could start with the default binary and use the estimators in phoebe to try to determine a starting point, but this still requires manually considering if the parameters that you are passing make sense and are physical.  At that point, you can use optimizers or samplers to fit around that region of parameter space to determine the best-fit parameters and their uncertainties to explain your observations.

**Reply by kittipongwangnok** [reply]
![Uploading Screenshot 2567-08-27 at 11.32.11 PM.png…]()
Thank you very much for your clarification. 
However, I just wondering, Can we reverse scale of y-axis like from small --> large as large ---> small with this command:

_ = b.plot(dataset='lc01', x='phases', 
           marker={'dataset': '.'},
           model='init_from_model', show=True)

**Reply by kecnry** [reply]
There are multiple ways to do that - have you tried anything that isn't working?

**Comment by kittipongwangnok** [comment]
Ok

Kittipong Wangnok
+6684 828 6505
School of Physics
Institute of Science
Suranaree University of Technology

**Comment by kittipongwangnok** [comment]
I already update it

Thanks
Kittipong

On Tue, Aug 27, 2024 at 9:03 PM Kyle Conroy ***@***.***>
wrote:

> All of your messages sent through email are being censored by GitHub, can
> you please go to the page and manually edit the messages so that they are
> readable?
>
> How can I enable the warning logger?
>
> #924 (comment)
> <https://github.com/phoebe-project/phoebe2/issues/924#issuecomment-2293424953>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/phoebe-project/phoebe2/discussions/928#discussioncomment-10463855>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALO4C3GWMXZ5EHPYTABPILLZTSBLRAVCNFSM6AAAAABNBGILP2VHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTANBWGM4DKNI>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Kittipong Wangnok
+6684 828 6505
School of Physics
Institute of Science
Suranaree University of Technology

**Comment by kittipongwangnok** [comment]
I was just wondering, can phoebe model the system with a period more than
300+ days or not?

On Wed, Aug 28, 2024 at 12:01 AM Kyle Conroy ***@***.***>
wrote:

> There are multiple ways to do that - have you tried anything that isn't
> working?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/phoebe-project/phoebe2/discussions/928#discussioncomment-10465812>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALO4C3HW4MQUHJGORUZLEOTZTSWHNAVCNFSM6AAAAABNBGILP2VHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTANBWGU4DCMQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Kittipong Wangnok
+6684 828 6505
School of Physics
Institute of Science
Suranaree University of Technology

**Comment by kittipongwangnok** [comment]
After I input:
b.set_value('period@binary@component', 310)

I got this error:
ValueError: Could not lookup ld_coeffs for ld_coeffs_source@primary@lc01@lc@dataset.  Try changing ld_coeffs_source to a table that covers a sufficient range of values or set ld_mode to 'manual' and manually provide coefficients via ld_coeffs. Enable 'warning' logger to see out-of-bound arrays.

**Reply by aprsa** [reply]
The exact same problem as before: you push your atmosphere parameters out of bounds by changing the orbital period. You will need to adjust other parameters accordingly so that Teff/logg/abun are within the supported range, or switch to blackbody.

**Comment by kittipongwangnok** [comment]
I have tried to check from ck2004 to blackbody, but I still get the error.
However, I have used the initial input file from "Time" to "Phase", then I
can model and get the model with the good fit but the default period in the
model is 1 day (but actually the real period of the model we use to convert
from time to phase already).
Can I model this way, we can interpret the other parameters?


On Sun, Sep 1, 2024 at 7:21 PM Andrej Prsa ***@***.***> wrote:

> The exact same problem as before: you push your atmosphere parameters out
> of bounds by changing the orbital period. You will need to adjust other
> parameters accordingly so that Teff/logg/abun are within the supported
> range, or switch to blackbody.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/phoebe-project/phoebe2/discussions/928#discussioncomment-10511963>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALO4C3GNVAKPVAJL3XLHUWLZUMBEVAVCNFSM6AAAAABNBGILP2VHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTANJRGE4TMMY>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Kittipong Wangnok
+6684 828 6505
School of Physics
Institute of Science
Suranaree University of Technology
